test/test_baggage.cpp

```diff
@@ -0,0 +1,214 @@
+#include <datadog/baggage.h>
+
+#include "catch.hpp"
+#include "mocks/dict_readers.h"
+#include "mocks/dict_writers.h"
+
+#define BAGGAGE_TEST(x) TEST_CASE(x, "[baggage]")
+
+using namespace datadog::tracing;
+
+BAGGAGE_TEST("missing baggage header is not an error") {
+  MockDictReader reader;
+  auto maybe_baggage = Baggage::extract(reader);
+  CHECK(!maybe_baggage);
+}
+
+BAGGAGE_TEST("extract") {
+  SECTION("limit is respected") {
+    const std::unordered_map<std::string, std::string> headers{
+        {"baggage", "team=proxy,company=datadog,user=dmehala"}};
+    MockDictReader reader(headers);
+
+    auto extraction_fails = Baggage::extract(reader, 1);
+    REQUIRE(!extraction_fails);
+    CHECK(extraction_fails.error() == Baggage::Error::MAXIMUM_CAPACITY_REACHED);
```

Comment:
According to our Baggage spec, we seem to only enforce baggage limits on inject and not at the time of extract so I think this should move this error to the injection logic

Reply:
Adressed in e3394c9f9e776b0f769c160914b30aeda6dea13a